test-framework/arquillian/src/main/java/io/quarkus/arquillian/QuarkusProtocol.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.arquillian;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.jboss.arquillian.container.spi.client.protocol.ProtocolDescription;
+import org.jboss.arquillian.container.spi.client.protocol.metadata.ProtocolMetaData;
+import org.jboss.arquillian.container.test.impl.client.protocol.local.LocalDeploymentPackager;
+import org.jboss.arquillian.container.test.impl.execution.event.LocalExecutionEvent;
+import org.jboss.arquillian.container.test.spi.ContainerMethodExecutor;
+import org.jboss.arquillian.container.test.spi.client.deployment.DeploymentPackager;
+import org.jboss.arquillian.container.test.spi.client.protocol.Protocol;
+import org.jboss.arquillian.container.test.spi.client.protocol.ProtocolConfiguration;
+import org.jboss.arquillian.container.test.spi.command.CommandCallback;
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.Injector;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.test.spi.TestMethodExecutor;
+import org.jboss.arquillian.test.spi.TestResult;
+
+import io.quarkus.arquillian.QuarkusProtocol.QuarkusProtocolConfiguration;
+
+public class QuarkusProtocol implements Protocol<QuarkusProtocolConfiguration> {
```

Comment:
Maybe it would be worth to review which classes we really need to make user-facing and narrow the scopes so they don't leak / explode when developing tests? WDYT?

Reply:
Makes sense.